github-pr-background.js

```diff
@@ -0,0 +1,95 @@
+"use strict";
+
+/**
+ * JS that runs in the background and has access to browser tabs.
+ */
+
+browser.runtime.onMessage.addListener(createAttachTab);
+
+/**
+ * Retrieve the currently active tab.
+ */
+function getActiveTab(callback) {
+    var allTabsPromise = browser.tabs.query({
+        currentWindow: true
+    });
+    // FIXME(willkg): browser.tabs.query returns an array of *strings* for the
+    // tab ids. This makes no sense--seems like it should return an array of
+    // Tabs.
+    allTabsPromise.then((tabs) => {
```

Comment:
A very nice feature that Firefox supports not is async/await. They are syntactic sugar around Promises that would let you write this as:

```js
async function getActiveTab() {
  const allTabs = await browser.tabs.query({ currentWindow: true });
  for (let tabId of allTabs) {
    // Convert the string tabId to an int and get that tab
    const tab = await browser.tabs.get(parseInt(tabId));
    if (tab.active) {
      return tab;
    }
}
```

That function will return a Promise, or can be used with await.